docs/js/doc-page.js

```diff
@@ -21,6 +21,94 @@ function removeDoubleSlashes(str) {
   return str.replace(/\/{2,}/g, '/');
 }
 
+function getPathFromSlug(str) {
+  return removeDoubleSlashes(`/${str}/`);
+}
+
+function relativeToAbsUrl(slug) {
+  return 'https://www.getambassador.io' + getPathFromSlug(slug);
+}
+
+const getDocPageSchema = ({
+  slug,
+  // an array of { title, slug } relative to parent pages (only if applicable)
+  // This doesn't include the root docs page nor the page itself
+  breadcrumbs = [],
+  // the doc page title (required)
+  title,
+  // the current docs version (required)
+  version,
+  // the current sidebar section (only if applicable)
+  section,
+  isFAQ,
+}) => ({
+  '@context': 'http://schema.org',
+  '@type': isFAQ ? 'FAQPage' : 'WebPage',
+  '@id': relativeToAbsUrl(slug),
+  // Not every search engine has assemblyVersion figured out (see below),
+  // we want to separate this doc page from others of different versions 😉
+  version,
```

Comment:
This will set the schema's `version` to the version of Ambassador, right? That wouldn't be quite right, would it? The version here should be the web-page's version, not the Ambassador release it describes, right?

Reply:
You mean `1.5.4` vs. `1.5`, for example? If so, absolutely agree!

How can we get the current web-page's version if the URL is `/latest`, though? With `/docs/1.4` and other numbered URLs it's pretty straightforward, but with `latest` I don't see a way around this.

Maybe we could get `versions.yml`'s `version` and get only the major and minor? In `1.5.4` we'd get only the `1.5` portion, for example. This would get complicated if there was ever any non-numbered version, though.

Thanks for your time, Luke!

Reply:
My final solution can be found here:
https://github.com/datawire/ambassador/blob/48ae642f45883fe4e68f84401ab6fe2190cf7701/docs/js/doc-page.js#L161-L163

What do you think??

Reply:
I meant the opposite of that.

I mean that
- The web page is describing 1.5.4
- But then someone improves the docs; not doing a release.
- That's a new "version" of the web page...
- ... which is what I understand `version` to mean at this place in the schema

That is: It is my understanding that the version here should be *more* granular than `versions.version`, not less.

(That understanding could very well be mistaken.)

Based on that understanding, I suspect the right thing to do would be to use `git describe --tags` as the doc version, but that's probably not worth the complexity at this point (maybe after we simplify how the versions get updated).

Reply:
Ohhh gotcha! We define the version in the schema in two different instances:

- the [`assemblyVersion`](https://schema.org/assemblyVersion) of the `APIReference`
  - according to Schema.org, `assemblyVersion` stands for _"Associated product/technology version. e.g., .NET Framework 4.5."_, so in this case it refers to the Ambassador version, not docs version
  - as we only have doc segments for minor releases (`1.5`) and not for patches ( `1.5.4`), I'm thinking only showing the minor version in `assemblyVersion` is the ideal, but this is open for debate :)
- the [`version`](https://schema.org/version) of the `WebPage` (or `FAQPage`)
  - according to Schema.org, this `version` represents _"The version of the CreativeWork embodied by a specified resource."_
  - so you're right, **it's the _version of the docs_, not the software**
  - that said, the goal here is mainly to show search engines there's a difference between the 1.5 and 1.4 docs version, but I don't think they're as granular in their understanding to make it worth the trouble of precisely explaining the docs version here
  - if we can't get to a solid answer, we could remove this property as the `@id` and the `mainEntity` of the pages are already different, let me know what you think

Thanks for spotting this nuance, @LukeShu !